### PR TITLE
ARGO-1217 Change service to service type

### PR DESCRIPTION
**Pull request by agelostsal**
Refactored service to service type, and updated tests, docs and swagger.

### CODE REVIEW

File: stores/mongo.go [hunk 1]
```diff
@@ -116,20 +116,20 @@ func (mongo *MongoStore) QueryBindings(service string, host string) ([]QBinding,
...
+func (mongo *MongoStore) InsertServiceType(name string, hosts []string, authTypes []string, authMethod string, retrievalField string, createdOn string) (QServiceType, error) {
 
-	var qService QService
+	var qService QServiceType
 	var err error
 
-	qService = QService{Name: name, Hosts: hosts, AuthTypes: authTypes, AuthMethod: authMethod, RetrievalField: retrievalField, CreatedOn: createdOn}
+	qService = QServiceType{Name: name, Hosts: hosts, AuthTypes: authTypes, AuthMethod: authMethod, RetrievalField: retrievalField, CreatedOn: createdOn}
 	db := mongo.Session.DB(mongo.Database)
 	c := db.C("services")
```
Comment (kaggis) [inline stores/mongo.go]: Refactoring resource name (services->service_types) should be reflected in data store as well for consistency

Comment (agelostsal) [reply stores/mongo.go]: Its been added in a later commit, but its true that it should be included here.

File: routing/routing.go [hunk 2]
```diff
@@ -62,10 +62,10 @@ func NewRouting(routes []APIRoute, store stores.Store, config *config.Config) *A
 }
 
 var ApiRoutes = []APIRoute{
-	{"services:create", "POST", "/services", handlers.ServiceCreate},
-	{"services:ListOne", "GET", "/services/{name}", handlers.ServiceListOne},
-	{"services:ListAll", "GET", "/services", handlers.ServiceListAll},
-	{"authMethod:ListOne", "GET", "/services/{service}/hosts/{host}/authM", handlers.AuthMethodListOne},
+	{"serviceTypes:create", "POST", "/service-types", handlers.ServiceTypeCreate},
```
Comment (kaggis) [inline routing/routing.go]: Can the path refactoring be reflected in handlers tests also for consistency?

Comment (agelostsal) [reply routing/routing.go]: yeah, fixing it now.

**Comment by kaggis** [comment]
@agelostsal looks great!

**Pull request merged**